src/TrackGenerator.cpp

```diff
@@ -423,13 +406,27 @@ FP_PRECISION TrackGenerator::getMaxOpticalLength() {
           sigma_t = material->getSigmaT();
 
           for (int e=0; e < material->getNumEnergyGroups(); e++)
-            max_optical_length = std::max(max_optical_length, length*sigma_t[e]);
+            max_optical_length = std::max(max_optical_length,
```

Comment:
Why isn't the max optical path length found in a kernel? Shouldn't all of this code iterating over tracks be abstracted into your kernel framework?